conan/tools/microsoft/visual.py

```diff
@@ -167,10 +169,23 @@ def generate(self, scope="build"):
                 Pop-Location
                 write-host conanvcvars.ps1: Activated environment}}
             """)
-            create_env_script(conanfile, content_ps1, CONAN_VCVARS_PS1, scope)
+            conan_vcvars_ps1 = f"{CONAN_VCVARS}.ps1"
+            create_env_script(conanfile, content_ps1, conan_vcvars_ps1, scope)
+            _create_deactivate_file(conanfile, conan_vcvars_ps1)
 
 
 
+def _create_deactivate_file(conanfile, filename):
```

Comment:
I'd name this something `vcvars` related, like
```suggestion
def _create_vcvars_deactivate_file(conanfile, filename):
```

as it only pertains to that